packages/e2e-tests/config/setup-screenshots.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const root = process.env.GITHUB_WORKSPACE || '../../../';
+const SCREENSHOTS_PATH = root + '/screenshots';
+
+if ( ! fs.existsSync( SCREENSHOTS_PATH ) ) {
+	fs.mkdirSync( SCREENSHOTS_PATH );
+}
+
+/**
+ * @copyright Tom Esterez (@testerez) https://github.com/smooth-code/jest-puppeteer/issues/131#issuecomment-424073620
+ */
+export const registerScreenshotReporter = () => {
+	/**
+	 * jasmine reporter does not support async.
+	 * So we store the screenshot promise and wait for it before each test
+	 */
+	let screenshotPromise = Promise.resolve();
+	beforeEach( () => screenshotPromise );
+	afterAll( () => screenshotPromise );
+
+	/**
+	 * Take a screenshot on Failed test.
+	 * Jest standard reporters run in a separate process so they don't have
+	 * access to the page instance. Using jasmine reporter allows us to
+	 * have access to the test result, test name and page instance at the same time.
+	 */
+	jasmine.getEnv().addReporter( {
+		specDone: async ( result ) => {
+			if ( result.status === 'failed' ) {
+				screenshotPromise = screenshotPromise
+					.catch()
+					.then( () => takeScreenshot( result.fullName ) );
```

Comment:
if `takeScreenshot` returns a promise (`page.screenshot`), could this just be `screenshotPromise = takeScreenshot( result.fullName )`

Curious why we have to `catch` the promise first 🤔

Reply:
I ended up removing this catch/then part entirely